validate-cert.js

```diff
@@ -6,10 +6,18 @@ module.exports = function validate(pem_cert) {
   try {
     var cert = pki.certificateFromPem(pem_cert)
 
-    // check that the domain echo-api.amazon.com is present in the Subject
-    // Alternative Names (SANs) section of the signing certificate
-    if (cert.subject.getField('CN').value.indexOf('echo-api.amazon.com') === -1) {
-      return 'subjectAltName Check Failed'
+    // check that cert has a Subject Alternative Names (SANs) section
+    var altNameExt = cert.getExtension("subjectAltName")
+    if (!altNameExt) {
+      return 'invalid certificate validity (subjectAltName extension not present)'
+    }
+
+    // check that the domain echo-api.amazon.com is present in SANs section
+    var domainExists = altNameExt.altNames.some(function(name) {
+      return name.value.indexOf('echo-api.amazon.com') >= 0
```

Comment:
I think this is problematic. simply checking to see if the cert name contains this string somewhere in it means people could register their own domains and get around this. For example, here is a site signed with a pem cert that would pass this check, but is owned by me, not amazon:  https://echo-api.amazon.com.reinstein.me/